packages/server/lib/cloud/protocol.ts

```diff
@@ -329,7 +348,7 @@ export class ProtocolManager implements ProtocolManagerShape {
 
     try {
       // @ts-expect-error - TS not associating the method & args properly, even though we know it's correct
-      await this._protocol[method].apply(this._protocol, args)
+      return await this._protocol[method].apply(this._protocol, args)
```

Comment:
if we are just yielding a promise here we should just be able to return the method like above
```suggestion
      return this._protocol[method].apply(this._protocol, args)
```

Reply:
I don't think this is the case actually. Since it's wrapped inside of a try catch log, the exceptions will go one level higher instead of in this method like we want.

```
(async function () { try { return new Promise((resolve, reject) => reject()) } catch (error) { console.log('caught') } })()
```

doesn't print 'caught' while

```
(async function () { try { return await new Promise((resolve, reject) => reject()) } catch (error) { console.log('caught') } })()
```

prints 'caught'

Reply:
ah so you want to yield the nested promise in that case. OK that makes sense